### PR TITLE
chore: reduce dist files to esm, cjs, and umd

### PR DESCRIPTION
**Pull request by 3cp**
BREAKING CHANGE: only distribute following files:
dist/meriyah.cjs
dist/meriyah.mjs
dist/meriyah.min.mjs
dist/meriyah.umd.js
dist/meriyah.umd.min.js
All transpiled with target: "ES2021"

### CODE REVIEW

**Comment by 3cp** [comment]
I locally tested it with `npm run bundle:bench` and `npm run bundle`.
I tried `npm pack` to produce a tgz file then consumed it in another project. Both CommonJS and ESM usages in Nodejs environment worked.

**Comment by 3cp** [comment]
Addresses #236

**Comment by fisker** [comment]
How about

```diff
- dist/meriyah.cjs.js
+ dist/meriyah.cjs
- dist/meriyah.cjs.min.js
- dist/meriyah.mjs
+ dist/meriyah.min.mjs
- dist/meriyah.umd.js
+ dist/meriyah.js
- dist/meriyah.umd.min.js
+ dist/meriyah.min.js
```

The`.cjs` `.mjs` extension already indentify what they are.

The CJS version can't use on web directly, I don't think it's nessary to minify.

File: package.json [hunk 1]
```diff
@@ -2,14 +2,13 @@
...
-  "jsnext:main": "dist/meriyah.esm.js",
+  "main": "dist/meriyah.cjs.js",
+  "module": "dist/meriyah.esm.mjs",
   "browser": "dist/meriyah.umd.min.js",
   "unpkg": "dist/meriyah.umd.min.js",
   "exports": {
     "import": "./dist/meriyah.esm.mjs",
-    "require": "./dist/meriyah.cjs",
+    "require": "./dist/meriyah.cjs.js",
     "types": "./dist/src/meriyah.d.ts"
```
Comment (fisker) [inline package.json:12]: I know current setting works , but TS  docs says

> The "types" condition should always come first in "exports".

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html#packagejson-exports-imports-and-self-referencing

Let's move it to top.

File: package.json [hunk 2]
```diff
@@ -31,6 +30,10 @@
     {
       "name": "Chunpeng Huo",
       "url": "https://github.com/3cp"
+    },
+    {
+      "name": "Fisker Cheung",
+      "url": "https://github.com/fisker"
```
Comment (fisker) [inline package.json:36]: I didn't contribute much to the source, I'd prefer not list myself here. I don't want see this field because huge in future.

Comment (3cp) [reply package.json:36]: OK :-)

**Comment by 3cp** [comment]
> How about
> 
> ```diff
> - dist/meriyah.cjs.js
> + dist/meriyah.cjs
> - dist/meriyah.cjs.min.js
> - dist/meriyah.mjs
> + dist/meriyah.min.mjs
> - dist/meriyah.umd.js
> + dist/meriyah.js
> - dist/meriyah.umd.min.js
> + dist/meriyah.min.js
> ```
> 
> The`.cjs` `.mjs` extension already indentify what they are.
> 
> The CJS version can't use on web directly, I don't think it's nessary to minify.

Good idea!
The `umd` might be worthy in the file name. When I wanted to manually use lib's dist file, I look those file names.

**Comment by fisker** [comment]
> The umd might be worthy in the file name. When I wanted to manually use lib's dist file, I look those file names.

Okay, also prevents link change on CDN.

File: package.json [hunk 3]
```diff
@@ -2,14 +2,13 @@
   "name": "meriyah",
   "version": "5.0.0",
   "description": "A 100% compliant, self-hosted javascript parser with high focus on both performance and stability",
-  "main": "dist/meriyah.cjs",
-  "module": "dist/meriyah.esm.js",
-  "jsnext:main": "dist/meriyah.esm.js",
+  "main": "dist/meriyah.cjs.js",
+  "module": "dist/meriyah.esm.mjs",
```
Comment (fisker) [inline package.json:6]: I think this `module` entry should link to minified version.

Import from  `https://unpkg.com/meriyah?module` will use it, jsdelivr should also use it I guess.

I think only CDNs and bundlers will look for this field.

Comment (3cp) [reply package.json]: Done.

**Review by fisker — APPROVED** [review]

**Comment by fisker** [comment]
Good job.

**Comment by fisker** [comment]
The branch is not deleted, maybe we can enable auto delete? https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-the-automatic-deletion-of-branches

**Comment by fisker** [comment]
@3cp 

Do you think add `"type": "module"` will break the dist files (I guess not)?
I'm trying to convert js files (in `/scripts`) to ESM. If it's not safe we can use `.mjs` instead.

**Comment by 3cp** [comment]
It will affect the umd.js file.
In `/scripts`, there is another package.json file to control the local "type".

**Comment by fisker** [comment]
> It will affect the umd.js file.

It's not in exports or main, so it's not possible to require or import. It can only loaded by `<script>`.

Of course ,  import it directly in our repo will throw, but this is only case I can image.

**Pull request merged**